### PR TITLE
GPU BasisApplyAtPoints

### PR DESCRIPTION
**Pull request by jeremylt**
~~Very WIP, blocked by #1595~~

I'm puzzling my way through exactly how the parallelism works.

Edit: Ready to rock

### CODE REVIEW

File: include/ceed/jit-source/cuda/cuda-ref-basis-tensor-at-points.h [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+        for (CeedInt d = 0; d < BASIS_DIM; d++) {
+          __syncthreads();
+          // Update buffers used
+          pre /= 1;
+          const CeedScalar *in       = d % 2 ? s_buffer_2 : s_buffer_1;
+          CeedScalar       *out      = d == BASIS_DIM - 1 ? s_chebyshev_coeffs : (d % 2 ? s_buffer_1 : s_buffer_2);
+          const CeedInt     writeLen = pre * post * Q;
+
+          // Build Chebyshev polynomial values
+          ChebyshevPolynomialsAtPoint<BASIS_Q_1D>(coords + elem * u_stride + d * u_comp_stride, chebyshev_x);
```
Comment (jeremylt) [inline include/ceed/jit-source/cuda/cuda-ref-basis-tensor-at-points.h:182]: I don't think this should be shared memory as its local to the point being evaluated?

Comment (jedbrown) [reply include/ceed/jit-source/cuda/cuda-ref-basis-tensor-at-points.h:182]: Yeah, anything local to the point would have no reason to be shared.

Comment (jeremylt) [reply include/ceed/jit-source/cuda/cuda-ref-basis-tensor-at-points.h:182]: Awesome, good to know I'm not fully crazy here

**Comment by jeremylt** [comment]
Most of the core pieces are there, need  to debug the parallelism, esp in 2/3d

**Comment by jeremylt** [comment]
This works on my machine, however on Noether I get the error
```
Test: t352-basis 
  $ build/t352-basis /gpu/cuda/ref
ERROR: returncode = 254
Output: 
NO MESSAGE
FAIL: stderr
Output: 
/var/lib/gitlab-runner/builds/N8GCsqus/0/libceed/libCEED/backends/cuda/ceed-cuda-compile.cpp:116 in CeedCompile_Cuda(): NVRTC_ERROR_COMPILATION
ptxas error   : Entry function 'GradAtPoints' uses too much shared data (0xcf90 bytes, 0xc000 max)
ptxas error   : Entry function 'InterpAtPoints' uses too much shared data (0xcf90 bytes, 0xc000 max)
make: *** [Makefile:697: junit-t352-basis] Error 1
```

**Comment by jeremylt** [comment]
Ok, fixed for now by shrinking the tests, they were using really large numbers of DoFs

**Pull request merged**